petab/core.py

```diff
@@ -38,87 +33,110 @@ class Problem:
     """
 
     def __init__(self,
-                 sbml_file,
-                 condition_file,
-                 measurement_file,
-                 parameter_file=None,
-                 model_name=None):
+                 sbml_model: libsbml.Model = None,
+                 condition_df: pd.DataFrame = None,
+                 measurement_df: pd.DataFrame = None,
+                 parameter_df: pd.DataFrame = None):
 
-        if model_name is None:
-            model_name = os.path.splitext(os.path.split(sbml_file)[-1])[0]
-        self.model_name = model_name
-
-        self.measurement_file = measurement_file
-        self.condition_file = condition_file
-        self.parameter_file = parameter_file
-        self.sbml_file = sbml_file
-
-        self.condition_df = None
-        self.measurement_df = None
-        self.parameter_df = None
-        self._load_dfs()
+        self.condition_df = condition_df
+        self.measurement_df = measurement_df
+        self.parameter_df = parameter_df
 
         self.sbml_reader = None
         self.sbml_document = None
-        self.sbml_model = None
-        self._load_sbml()
+        self.sbml_model = sbml_model
```

Comment:
it said somewhere at some point that the sbml_read and _document must be kept in memory, since otherways future calls will fail (don't know the details). just to remark this here.

Reply:
Yes, it did. User-responsibility. We don't care about document and reader, only if we open it ourselves. This restriction will also be gone with the next libsbml release.

Reply:
Now you can also pass everything to Problem to not have to care about it yourself.